test/terra/backends/aer_simulator/test_fusion.py

```diff
@@ -511,3 +512,45 @@ def test_fusion_diagonal(self):
             if op_name == 'measure':
                 break
             self.assertEqual(op_name, 'diagonal')
+
+
+    def test_parallel_fusion_diagonal(self):
+        """Test diagonal fusion with parallelization"""
+        backend = self.backend(method="statevector")
+        
+        num_of_qubits = 10
+        circuit = QuantumCircuit(num_of_qubits)
+        size = 3
+        
+        for q in range(num_of_qubits):
+            circuit.h(q)
+        
+        for qubit in range(num_of_qubits):
+            for q in range(size):
+                ctrl = (qubit + q) % num_of_qubits
+                tgt = (ctrl + 1) % num_of_qubits
+                circuit.cx(ctrl, tgt)
+        
+            circuit.p(np.random.random(), tgt)
```

Comment:
It'd be best to make this a fixed value, rather than making the test stochastic - if it were to ever break, it'd be impossible to reproduce.